src/service.rs

```diff
@@ -32,25 +34,10 @@ pub struct BitcoinService {
     client: BitcoinNode,
     rollup_name: String,
     network: bitcoin::Network,
-    address: String,
+    address: Address<NetworkUnchecked>,
```

Comment:
Why unchecked if we are forcing the network

Reply:
Network checked cannot be serialized so the struct has unchecked but before use it enforces the network (panics if fails)

Reply:
ok then store it as unchecked but make the check in with_client please not before sending